pvm.py

```diff
@@ -95,68 +100,98 @@ def parse_commands(*args):
 			return
 
 		if command=="start":
-			if media.can_play():
-				media.play()
+			if CAN_START:
+				OMX.play()
+				CAN_START = False
+				CAN_PAUSE = True
 				_logger.info("%s command success.", command)
 			else:
 				_logger.info("%s command failed.", command)
-		elif command=="stop":
-			if media.can_quit():
-				media.stop()
-				IS_FILE_SET = False
-				_logger.info("%s command success and file has been unset.", command)
+		elif command=="pause":
+			if CAN_PAUSE:
+				OMX.pause()
+				CAN_PAUSE = False
+				CAN_START = True
+				_logger.info("%s command success.", command)
 			else:
 				_logger.info("%s command failed.", command)
+		elif command=="stop":
+			OMX.stop()
+			IS_FILE_SET = False
+			CAN_START = False
+			CAN_PAUSE = False
+			_logger.info("%s command success and %s has been unset.", command, VIDEO_PATH)
 		elif command=="set_position":
-			media.set_position(float(value))
+			OMX.set_position(float(value))
 			_logger.info("%s command success.", command)
 		elif command=="set_rate":
-			fps = str(30 * float(value))
-			media = OMXPlayer(PREFIX_PATH + VIDEO_PATH, dbus_name='org.mpris.MediaPlayer2.omxplayer', args=['--loop','--force-fps', fps])
-			if media != "":
-				media.quit()
-				print("media quit")
-			original_fps = 30
-			video_info = subprocess.Popen(["omxplayer", "-i", PEFIX_PATH + VIDEO_PATH], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-			print("The command", ["omxplayer", "-i", PEFIX_PATH + VIDEO_PATH])
-			# info = video_info.stdout.decode().split(", ")
-			out, err = video_info.communicate()
-			out = out.decode(encoding='utf-8')
-			splist = out.split(", ")
-			for s in splist:
-				if 'fps' in s:
-					print("current ele:", s)
-					fps_info = s.split(' ')
-					original_fps = float(fps_info[0])
-			fps = str(original_fps * float(value))
-			print("computed fps", fps)
-			media = OMXPlayer(PREFIX_PATH + VIDEO_PATH, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=['--loop','--force-fps', fps])
-			media.pause()
-			_logger.info("%s command success.", command)
-		elif command=="pause":
-			if media.can_pause():
-				media.pause()
-				_logger.info("%s command success.", command)
-			else:
-				_logger.info("%s command failed.", command)
+			fps = get_rate(value)
+			if IS_FILE_SET:
+				OMX.quit()
+			params = params + ['--force-fps', fps]
+			OMX = OMXPlayer(VIDEO_PATH, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=params)
+			OMX.pause()
+			CAN_START = True
+			CAN_PAUSE = False
+			_logger.info("%s command success set rate: %s. You can start now!", command, fps)
```

Comment:
oh, so changing the rate, requires reloading the file? that is really bad news.
Our goal was to change the rate interactively as the video is playing... make it go faster or slower.
I will be really surprised if that is not possible to achieve, it is a very simple effect, right?

Reply:
Done, please see the commit [here](https://github.com/KaneBetter/omxplayer/commit/9474c053dfcbe6dde6da34593880334f4d1566f9).